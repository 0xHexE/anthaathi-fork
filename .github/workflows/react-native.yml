name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
