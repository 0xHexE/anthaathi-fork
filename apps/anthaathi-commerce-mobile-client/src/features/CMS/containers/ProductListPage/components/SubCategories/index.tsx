import {View, VirtualizedList} from 'react-native';
import React from 'react';
import {Button} from 'react-native-paper';
import {ProductListPageComponentType} from '../../../../types/common';

type CategoryDetailsProps = {
  id: string;
  title: string;
};

export interface SubCategoriesProps {
  subCategoryList: CategoryDetailsProps[];
}

const SubCategories = (props: SubCategoriesProps) => {
  return (
    <View style={{marginVertical: 5}} testID="subCategories">
      <VirtualizedList<CategoryDetailsProps>
        testID="subCategoryList"
        data={props.subCategoryList}
        initialNumToRender={4}
        horizontal={true}
        renderItem={({item}) => <ItemRenderer item={item} />}
        getItemCount={() => props.subCategoryList.length}
        keyExtractor={item => item.id}
        getItem={(res, index) => res[index]}
      />
    </View>
  );
};

const ItemRenderer = ({item}: {item: CategoryDetailsProps}) => {
  return (
    <Button
      mode={item.id === '1' ? 'contained' : 'outlined'}
      style={{
        marginHorizontal: 5,
        marginVertical: 2,
        borderColor: '#E3E2E7',
        borderRadius: 2,
      }}
      labelStyle={{
        fontSize: 12,
        fontWeight: '400',
        color: item.id === '1' ? '#ffffff' : '#364A15',
      }}
      onPress={() => console.log('Pressed')}>
      {item.title}
    </Button>
  );
};

export default SubCategories;

export const SubCategoriesCMSInput = {
  _component: ProductListPageComponentType.SubCategories,
  component: SubCategories,
};
