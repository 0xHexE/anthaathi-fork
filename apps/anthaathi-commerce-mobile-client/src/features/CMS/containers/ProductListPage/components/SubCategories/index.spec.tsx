import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import SubCategories from '.';

describe('SubCategories', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SubCategories
            subCategoryList={[
              {
                id: '1',
                title: 'All Fruits',
              },
              {
                id: '2',
                title: 'All Fruits',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('subCategories')).toBeTruthy();
  });

  it('renders expected number of products', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SubCategories
            subCategoryList={[
              {
                id: '1',
                title: 'All Fruits',
              },
              {
                id: '2',
                title: 'All Fruits',
              },
            ]}
          />{' '}
        </IntlProvider>
      </ThemeProvider>,
    );

    const getTestId = temp.queryByTestId('subCategoryList')!;
    expect(getTestId.props.data.length).toBe(2);
    expect(getTestId.props.data[0]).toEqual({
      id: '1',
      title: 'All Fruits',
    });
  });
});
