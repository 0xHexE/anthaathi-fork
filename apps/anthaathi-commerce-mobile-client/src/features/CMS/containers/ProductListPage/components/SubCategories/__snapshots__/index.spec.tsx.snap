// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategories should render the item 1`] = `
<View
  style={
    Object {
      "marginVertical": 5,
    }
  }
  testID="subCategories"
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1",
          "title": "All Fruits",
        },
        Object {
          "id": "2",
          "title": "All Fruits",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={4}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="subCategoryList"
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(103, 80, 164, 1)",
                  "borderColor": "#E3E2E7",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "marginHorizontal": 5,
                  "marginVertical": 2,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 2,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          Object {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          Object {
                            "color": "rgba(255, 255, 255, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Object {
                            "color": "#ffffff",
                            "fontSize": 12,
                            "fontWeight": "400",
                          },
                        ],
                      ]
                    }
                  >
                    All Fruits
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#E3E2E7",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "marginHorizontal": 5,
                  "marginVertical": 2,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 2,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          Object {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          Object {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Object {
                            "color": "#364A15",
                            "fontSize": 12,
                            "fontWeight": "400",
                          },
                        ],
                      ]
                    }
                  >
                    All Fruits
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
